### DISCUSSION TITLE
Where is the Scaladoc for the core mill packages?

### DISCUSSION THREAD

**Discussion by Sailsman63** [com-lihaoyi/mill#1688] [Q&A]
As a total mill beginner/scala newbie, I've been finding myself searching through the core repository here trying to understand what tasks are available out-of-the-box, and what options or configuration for them are available. Is the Scaladoc/javadoc for them hosted somewhere, and I just missed it?

**Comment by lefou** [comment]
The probably easiest way to browse mill API is by opening a `build.sc` in a supported IDE (either IntelliJ IDEA or any BSP client). Then these fetch all required source jars in the background and you get very good editing and browsing support, like inspecting the various Mill modules, but also auto completion.

Mills API is partitioned into various jars and respectively source jars. You can already browse these online, e.g. via javadoc.io. Example for current scalalib documentation: https://javadoc.io/doc/com.lihaoyi/mill-scalalib_2.13/0.10.0/index.html

I'd like to also have a combined Scaladoc site, but this is a work in progress: https://github.com/com-lihaoyi/mill/pull/1559

**Reply by Sailsman63** [reply]
Thank you for the information.

I'm finding myself a  bit disappointed here, on a couple of fronts:

- javadoc.io is essentially useless here, as I don't even know which jars I should be looking at in the first place. I get that it's automatically generated, and my use case is probably outside what it's intended for.
- I don't use IDEs, and even if I did, I don't think auto-complete support and friends are all that useful when I don't know roughly what I'm looking for in the first place.
- Reviewing #1559, I'm a bit surprised that there isn't a combined-doc type task available as standard in the first place. I certainly think that it would be something that projects would not want to be continuously reinventing. Would this be worth a feature request, do you think?

One thing that I have been liking is the accessibility of the beginner docs. They are a major contrast to trying to get started with Gradle or Maven, for example.

**Reply by lefou** [reply]
Yeah, some kind of unidoc support or plugin for Mill would be really nice and would help to provide a single go to resource where one can explore Mill API. As a user who wants to get a more comprehensible overview, some kind of module reference would be even more helpful, but might take even more effort (initially and in maintenance).
